File: BloodBank/src/main/java/com/example/BloodBank/model/Questionnaire.java
package com.example.BloodBank.model;

import com.example.BloodBank.dto.QuestionnaireDTO;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import java.io.Serializable;
import java.time.LocalDate;

@Entity()
@NoArgsConstructor
@AllArgsConstructor
@Data
public class Questionnaire implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    @ManyToOne(cascade = CascadeType.MERGE)
    @JoinColumn(name = "customer_id", referencedColumnName = "id")
    private Customer customer;

//    @NotNull
//    @NotBlank
    private Boolean donated;
    private Boolean denied;
    private Boolean healthy;
    private Boolean eaten;
    private Boolean dangerousJob;
    //FEMALES ONLY
    private Boolean secondState;
    private Boolean menstruating;
    private Boolean pregnant;
    private String donorNumber;
    private LocalDate fillDate;

    public boolean checkIfValidForReservationOfAppointment() {
        return fillDate.isAfter(LocalDate.now().minusMonths(1));
    }

}

File: ISAfrontend/src/app/modules/appointments/model/questionnaire.ts
export class Questionnaire {
    id: string = "-1";
    customerId : string = "-1";
    donated: boolean;
    denied: boolean;
    healthy: boolean;
    eaten: boolean;
    dangerousJob: boolean;
    takingMedicine: boolean;
    secondState: boolean;
    menstruating: boolean;
    pregnant: boolean;
    donorNumber: string = '';
    fillDate: Date = new Date();

    public constructor(obj?: any) {
        this.id = obj?.id ?? "-1";
        this.customerId = obj?.customerId ?? "-1";
        this.donated = obj?.donated
        this.denied = obj?.denied
        this.healthy = obj?.healthy
        this.eaten = obj?.eaten
        this.dangerousJob = obj?.dangerousJob
        this.takingMedicine = obj?.takingMedicine
        this.secondState = obj?.secondState
        this.menstruating = obj?.menstruating
        this.pregnant = obj?.pregnant
        this.donorNumber = obj?.donorNumber ?? '';
        this.fillDate = obj?.fillDate ?? new Date();
    }
}
